packages/autorest.go/src/generator/constants.ts

```diff
@@ -3,55 +3,52 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { Session } from '@autorest/extension-base';
 import { comment } from '@azure-tools/codegen';
-import { CodeModel, ChoiceValue, Schemas } from '@autorest/codemodel';
-import { length, values } from '@azure-tools/linq';
-import { contentPreamble, hasDescription, sortAscending } from './helpers';
-import { commentLength } from '../common/helpers';
+import { values } from '@azure-tools/linq';
+import { commentLength, contentPreamble } from './helpers';
+import { GoCodeModel } from '../gocodemodel/gocodemodel';
 
 // Creates the content in constants.go
-export async function generateConstants(session: Session<CodeModel>): Promise<string> {
-  const enums = getEnums(session.model.schemas);
+export async function generateConstants(codeModel: GoCodeModel): Promise<string> {
   // lack of operation groups indicates model-only mode.
-  if (length(session.model.operationGroups) === 0 || (enums.length === 0 && !session.model.language.go!.host && !session.model.language.go!.azureARM)) {
+  if (!codeModel.clients || (!codeModel.constants && !codeModel.host && codeModel.type !== 'azure-arm')) {
     return '';
   }
-  let text = await contentPreamble(session);
-  if (session.model.language.go!.host) {
-    text += `const host = "${session.model.language.go!.host}"\n\n`;
+  let text = contentPreamble(codeModel);
+  if (codeModel.host) {
+    text += `const host = "${codeModel.host}"\n\n`;
   }
   // data-plane clients must manage their own constants for these values
-  if (<boolean>session.model.language.go!.azureARM) {
-    if (session.model.language.go!.moduleVersion === '') {
+  if (codeModel.type === 'azure-arm') {
```

Comment:
We do this check is a few places, Should we make this a helper function?  Or at a minimum make `azure-arm` a constant.

Reply:
The value is a discriminated union, see [here](https://github.com/Azure/autorest.go/blob/6b12a984bfccc7093f342d7757889ec12d09a75c/packages/autorest.go/src/gocodemodel/gocodemodel.ts#L40).

Reply:
Makes sense, Why not wrap that into a function to avoid having `azure-arm` written throughout code?
if(codeModel.isAzureARM())

Reply:
Here, `codeModel` is an interface, not a class, so it can't have any methods.  There might be some magic to make this work.  Will revisit in a separate PR.